advanced-cache.php

```diff
@@ -642,8 +649,10 @@ function add_debug_html_to_output( $debug_html ) {
 
 	// Use the batcache save time for Last-Modified so we can issue "304 Not Modified" but don't clobber a cached Last-Modified header.
 	if ( $batcache->cache_control && !isset($batcache->cache['headers']['Last-Modified'][0]) ) {
+		$max_age = ( $batcache->cache['max_age'] - time() + $batcache->cache['time'] );
+		$max_age = ( $max_age >= $batcache->max_age_stale ) ? $max_age : $batcache->max_age_stale;
```

Comment:
Hmm I think this isn't in the context of knowing if this is a stale cache item, so effectively it's just making any cache older that `max_age_stale` to be that value? That's what I'm reading this as anyway.

Reply:
We might need to re-do this check at https://github.com/humanmade/batcache/pull/10/files#diff-16af6f433245dc1d68537817163be874R591 which is where the "use stale in this case" detection is done.

Reply:
So this is where I don't totally understand the problem. If the requests were serving a negative max-age, they _had_ to have reached this point in the file (being these lines doing the arithmetic), meaning _also_ they were being served out of the cache just a few lines below this on 685. Is that right? How would there be a negative max-age but not serve it from the cache?

Reply:
Right, you are correct, but with the change now, anything `< $max_age_stale` will be set to `$max_age_stale`. E.g. if the cache has 5 seconds left to live, this is going to set it to 10. Because there's no check here to say "If this is a stale cache, set it to `max_age_stale`. 

This should be something more like `$max_age = $max_age > 0 ?: $batcache->max_age_stale;`

I.e. "If max age is 0 or less, we know it must be stale, so set the max_age to the max_age_stale`.

Reply:
Got it. Added!